plugins/commands/remove.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+const boom = require('boom');
+const joi = require('joi');
+const schema = require('screwdriver-data-schema');
+const baseSchema = schema.models.command.base;
+
+module.exports = () => ({
+    method: 'DELETE',
+    path: '/commands/{namespace}/{name}',
+    config: {
+        description: 'Delete a command',
+        notes: 'Returns null if successful',
+        tags: ['api', 'commands'],
+        auth: {
+            strategies: ['token'],
+            scope: ['build', 'user', '!guest']
+        },
+        plugins: {
+            'hapi-swagger': {
+                security: [{ token: [] }]
+            }
+        },
+        handler: (request, reply) => {
+            const { namespace, name } = request.params;
+            const { credentials } = request.auth;
+            const { commandFactory, commandTagFactory } = request.server.app;
+            const { canRemove } = request.server.plugins.commands;
+
+            return Promise.all([
+                commandFactory.list({ params: { namespace, name } }),
+                commandTagFactory.list({ params: { namespace, name } })
+            ]).then(([commands, tags]) => {
+                if (commands.length === 0) {
+                    throw boom.notFound(`Command ${namespace}/${name} does not exist`);
+                }
+
+                return canRemove(credentials, commands[0], 'admin').then(() => {
+                    const commandPromises = commands.map(command => command.remove());
+                    const tagPromises = tags.map(tag => tag.remove());
+
+                    return Promise.all(commandPromises.concat(tagPromises));
```

Comment:
I think you should also remove the command binaries from the store. So you'll have to make some call to the store similar to https://github.com/screwdriver-cd/screwdriver/blob/master/plugins/commands/create.js#L43-L51
But I don't think that endpoint exists yet in the store (https://github.com/screwdriver-cd/store/blob/master/plugins/commands.js). You might want to coordinate with Yahoo Japan. Not sure if they're already working on this.